haystack/nodes/prompt/prompt_node.py

```diff
@@ -0,0 +1,822 @@
+        template_dict = {}
+        # attempt to resolve args first
+        if args:
+            if len(args) != len(self.prompt_params):
+                logger.warning(
+                    f"For {self.name}, expected {self.prompt_params} arguments, instead "
+                    f"got {len(args)} arguments {args}"
+                )
+            for prompt_param, arg in zip(self.prompt_params, args):
+                template_dict[prompt_param] = [arg] if isinstance(arg, str) else arg
+        # then attempt to resolve kwargs
+        if kwargs:
+            for param in self.prompt_params:
+                if param in kwargs:
+                    template_dict[param] = kwargs[param]
+
+        if set(template_dict.keys()) != set(self.prompt_params):
+            available_params = set(list(template_dict.keys()) + list(set(kwargs.keys())))
+            raise ValueError(f"Expected prompt params {self.prompt_params} but got {list(available_params)}")
+
+        template_dict["prompt_template"] = self.prompt_text
+        return template_dict
+
+
+PREDEFINED_PROMPT_TEMPLATES = [
+    PromptTemplate(
+        name="question-answering",
+        prompt_text="Given the context please answer the question. Context: $documents; Question: $questions; Answer:",
+        prompt_params=["documents", "questions"],
+    ),
+    PromptTemplate(
+        name="question-generation",
+        prompt_text="Given the context please generate a question. Context: $documents; Question:",
+        prompt_params=["documents"],
+    ),
+    PromptTemplate(
+        name="conditioned-question-generation",
+        prompt_text="Please come up with a question for the given context and the answer. "
+        "Context: $documents; Answer: $answers; Question:",
+        prompt_params=["documents", "answers"],
+    ),
+    PromptTemplate(
+        name="summarization", prompt_text="Summarize this document: $documents Summary:", prompt_params=["documents"]
+    ),
+    PromptTemplate(
+        name="question-answering-check",
+        prompt_text="Does the following context contain the answer to the question. "
+        "Context: $documents; Question: $questions; Please answer yes or no! Answer:",
+        prompt_params=["documents", "questions"],
+    ),
+    PromptTemplate(
+        name="sentiment-analysis",
+        prompt_text="Please give a sentiment for this context. Answer with positive, "
+        "negative or neutral. Context: $documents; Answer:",
+        prompt_params=["documents"],
+    ),
+    PromptTemplate(
+        name="multiple-choice-question-answering",
+        prompt_text="Question:$questions ; Choose the most suitable option to answer the above question. "
+        "Options: $options; Answer:",
+        prompt_params=["questions", "options"],
+    ),
+    PromptTemplate(
+        name="topic-classification",
+        prompt_text="Categories: $options; What category best describes: $documents; Answer:",
+        prompt_params=["documents", "options"],
```

Comment:
Should we use `categories` as the variable name to be consistent with how the prompt is structured?

Also should `prompt_params` be updated to 
```python
prompt_params=["options", "documents"]
```
since `options` comes before `documents` in the prompt?